File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/src (BUX)/View/TextMenu.java
package View;

import View.Commands.Command;

import java.util.*;


public class TextMenu
{
    private Map<String, Command> commands;
    public TextMenu(){commands = new HashMap<>();}
    public void addCommand(Command c){commands.put(c.getKey(),c);}
    private void printMenu()
    {
        for(Command com:commands.values())
        {
            String line = String.format("%4s: %s", com.getKey(), com.getDescription());
            System.out.println(line);
        }
    }

    public void show()
    {
        Scanner scanner = new Scanner(System.in);
        while(true)
        {
            printMenu();
            System.out.println("Choose option: ");
            String key = scanner.nextLine();
            Command com = commands.get(key);
            if(com == null)
            {
                System.out.println("Invalid option!");
                continue;
            }
            com.execute();
        }
    }

}

File: 3rd Semester/Advanced Programming Methods/Laboratory/Toy Interpreter C#/sharplab/View/TextMenu.cs
using sharplab.View.Commands;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace sharplab.View
{
    class TextMenu
    {
        private IDictionary<string, Command> commands;
        public TextMenu() { commands = new Dictionary<string,Command>(); }
        public void AddCommand(Command c) { commands.Add(c.GetName(), c); }
        private void PrintMenu()
        {
            foreach (Command com in commands.Values.ToArray())
            {
                string line = com.GetName() + "." + com.GetDescription();
                Console.WriteLine(line);
            }
        }

        public void Show()
        {
            while (true)
            {
                PrintMenu();
                System.Console.WriteLine("Choose option: ");
                String key = System.Console.ReadLine();
                Command com = commands[key];
                if (com == null)
                {
                    System.Console.WriteLine("Invalid option!");
                    continue;
                }
                com.Execute();
            }
        }

    }
}
